app-backend/batch/src/main/scala/com/azavea/rf/batch/airflow/export/CreateExportDef.scala

```diff
@@ -0,0 +1,178 @@
+import com.amazonaws.services.route53._
+import com.amazonaws.services.route53.model._
+import io.circe.generic.JsonCodec
+import io.circe.syntax._
+
+import com.azavea.rf.batch.JobNoEC
+import com.azavea.rf.batch.util._
+import com.azavea.rf.database.{Database => DB}
+import com.azavea.rf.database.tables._
+import com.azavea.rf.datamodel._
+
+case class CreateExportDef(exportId: UUID)(implicit val database: DB) extends JobNoEC {
+  val name = "create_export_def"
+  implicit val system       = ActorSystem(s"$name-system")
+  implicit val materializer = ActorMaterializer()
+    
+  protected def writeExportDefToS3(exportDef: ExportDefinition) = {
+    logger.info(s"Uploading export definition ${exportDef.id.toString} to S3 at ${exportDefConfig.bucketName}")
+    val result = Either.catchNonFatal(
+      S3.putObject(s"rasterfoundry-development-data-us-east-1/${exportDefConfig.bucketName}", s"${exportDef.id.toString}.json", None, exportDef.asJson.toString))
+    result match {
+      case Right(_) => s"rasterfoundry-development-data-us-east-1/${exportDefConfig.bucketName}"
+      case Left(ex) => {
+        logger.info(ex.toString())
+        throw new Exception("...")
+      }
+    }
+  }
+
+  def wrapTry[A,B](fn: A => B): A => Option[B] = {
+    (x) => {
+      val result = Either.catchNonFatal(fn(x))
+      result match {
+        case Right(r) => r.some
+        case Left(ex) => {
+          logger.error(ex.toString())
+          None
+        }
+      }
+    }
+  }
+
+  def fx[A,B](fn: A => B): A => Option[A] = {
+    (x) => {
+      wrapTry((y:A) => fn(y)).apply(x) match {
+       case Some(_) => x.some
+       case None => None
+      }
+    }
+  }
+
+  protected def getClusterId() = {
+    val r53id = "ZBSHT5XZMJFNP"
+    val r53Client = (
+      ().some
+        >>= wrapTry((_:Unit) => new DefaultAWSCredentialsProviderChain())
+        >>= wrapTry(_.getCredentials())
+        >>= wrapTry(cred => new AmazonRoute53Client(cred))
+        >>= fx(_ => logger.info("Created Route 53 client"))
+    ).getOrElse(throw new Exception("Could not instantiate Route 53 client"))
+
+    val result = (
+      r53Client.some
+        >>= wrapTry(_.listResourceRecordSets(new ListResourceRecordSetsRequest(r53id)))
+        >>= fx(_ => logger.info(s"Retrieved ResourceRecordSets for ${r53id}"))
+        >>= wrapTry(_.getResourceRecordSets())
+        >>= wrapTry(_.asScala.toList)
+        >>= wrapTry(_.filter(r => r.getName() == "dataproc.rasterfoundry.com.").head)
+        >>= wrapTry(_.getResourceRecords())
+        >>= wrapTry(_.asScala.toList.head)
+        >>= wrapTry(_.getValue())
+        >>= fx(c => logger.info(s"Using cluster ${c}"))
+        >>= wrapTry(_.replace("\"", ""))
+    ).getOrElse(throw new Exception("Could not get Spark cluster"))
+
+    result
+  }
+
+  protected def startExportEmrJob(exportDef: ExportDefinition, exportDefUri: String): Unit = {
+    val jarStep = (
+      ().some
+        >>= wrapTry((_:Unit) => new HadoopJarStepConfig())
```

Comment:
Ah ok, I think it is coming in from `cats`.

I think it would be good to do something more conventional that fits in with our project's style more and since the usage here isn't gaining us much.

E.g. something like:
```scala
val stepConfig = new HadoopJarStepConfig()
stepConfig.setJar(jarPath)
stepConfig.setArgs(<etc>)
...
```